### PR TITLE
📥 Prefer frontmatter downloads over exports

### PR DESCRIPTION
**Pull request by fwkoch**
Here's a start to consume the new `downloads` available in project/page frontmatter (https://github.com/executablebooks/mystmd/pull/1049)

Where we previously built a download list from `exports` we now look to see if `downloads` is defined. If so, the theme uses that instead. If `downloads` is not defined, it still falls back to `exports` for backwards compatibility.

This also makes a couple tweaks so download `format` may be undefined - since we allow downloads to be of arbitrary format, we need to handle unknown formats.

### CODE REVIEW

File: themes/article/app/components/Downloads.tsx [hunk 1]
```diff
@@ -28,9 +32,10 @@ export function Downloads() {
             key={action.url}
             href={action.url}
             className="inline-block mr-2 no-underline lg:mr-0 lg:block"
+            target={action.internal ? '_self' : '_blank'}
```
Comment (fwkoch) [inline themes/article/app/components/Downloads.tsx:35]: There's probably a better way to do this...?

Comment (fwkoch) [reply themes/article/app/components/Downloads.tsx:35]: This also isn't actually respected for relative urls.

**Comment by stevejpurves** [comment]
This PR so far addresses the `article-theme`, how should the `book-theme` be handling these given they are defined on the project?

File: themes/article/app/components/Downloads.tsx (hunk 1 above)
Comment (stevejpurves) [reply themes/article/app/components/Downloads.tsx]: changed this up significantly

File: themes/article/app/components/Downloads.tsx [hunk 2]
```diff
@@ -15,24 +23,83 @@ function formatToTitle(format: string) {
...
-          </a>
-        ))}
+          );
+
+          return (
+            <a
+              key={item.url}
+              href={item.url}
+              className="inline-block mr-2 no-underline lg:mr-0 lg:block"
+              download={item.static ? item.filename : undefined}
```
Comment (fwkoch) [inline themes/article/app/components/Downloads.tsx:91]: @fwkoch  Change these to be based on `filename`, not `static`

**Comment by fwkoch** [comment]
I've aligned `DownloadsDropdown` in `frontmatter` with your changes to the article theme - this addresses book theme since it uses that component.

The only difference is `frontmatter` does not depend on `@remix-run/react` so it does not use `Link` for the internal links. Not sure if there's any implications to adding that dep to `frontmatter`?

**Comment by stevejpurves** [comment]
@fwkoch yeah let's keep `Link` out of frontmatter and just use `a` there

**Review by stevejpurves — APPROVED** [review]
tested both themes out locally, looks great @fwkoch!

**Pull request merged**